sdk/program/src/stake/state.rs

```diff
@@ -567,13 +567,31 @@ impl Stake {
         Ok(new)
     }
 
-    pub fn deactivate(&mut self, epoch: Epoch) -> Result<(), StakeError> {
+    pub fn deactivate(
+        &mut self,
+        epoch: Epoch,
+        stake_flags: &StakeFlags,
+        history: Option<&StakeHistory>,
+    ) -> Result<(), StakeError> {
         if self.delegation.deactivation_epoch != std::u64::MAX {
-            Err(StakeError::AlreadyDeactivated)
-        } else {
-            self.delegation.deactivation_epoch = epoch;
-            Ok(())
+            return Err(StakeError::AlreadyDeactivated);
```

Comment:
nit: generally prefer to scope explicitly (especially in the native programs) rather than early return. early returns make introducing result priority reordering bugs far too easy

Reply:
done